packages/dappmanager/hostScripts/docker_update.sh

```diff
@@ -0,0 +1,343 @@
+    ARCHITECTURE=$(uname -m)
+    # x86_64 is equal to amnd64, correct the value
+    if [[ "$ARCHITECTURE" == "x86_64" ]]; then
+      ARCHITECTURE="amd64"
+    fi
+  else
+    echo "Error retrieving architecture"
+    exit 3
+  fi
+}
+
+function install_docker_engine(){
+  # Note: The client and container runtime are now in separate packages from the daemon in Docker Engine 18.09.
+  # Users should install and update all three packages at the same time to get the latest patch releases
+
+  # TEMPORARY: Bullseye consider as Buster until it gets supported by docker
+  if [ "$VERSION_CODENAME" == "bullseye" ]; then
+    VERSION_CODENAME="buster"
+  fi
+
+  # Change versions for stretch
+  if [ "$VERSION_CODENAME" != "buster" ]; then
+    STABLE_DOCKER_ENGINE_VERSION="19.03.8"
+    STABLE_DOCKER_CONTAINERD_VERSION="1.2.6-3"
+  fi
+  
+  # Metadata
+  DOCKER_PKG="docker-ce_${STABLE_DOCKER_ENGINE_VERSION}~3-0~debian-${VERSION_CODENAME,,}_${ARCHITECTURE}.deb"
+  DOCKER_CLI="docker-ce-cli_${STABLE_DOCKER_ENGINE_VERSION}~3-0~debian-${VERSION_CODENAME,,}_${ARCHITECTURE}.deb"
+  DOCKER_CONTAINERD="containerd.io_${STABLE_DOCKER_CONTAINERD_VERSION}_${ARCHITECTURE}.deb"
+
+  # URLs
+  DOCKER_URL="https://download.docker.com/linux/debian/dists/${VERSION_CODENAME,,}/pool/stable/${ARCHITECTURE}"
+  DOCKER_PKG_URL="${DOCKER_URL}/${DOCKER_PKG}"
+  DOCKER_CLI_URL="${DOCKER_URL}/${DOCKER_CLI}"
+  DOCKER_CONTAINERD_URL="${DOCKER_URL}/${DOCKER_CONTAINERD}"
+
+  # Paths
+  DOCKER_ENGINE_PATH="/usr/src/dappnode"
+  DOCKER_PKG_PATH="${DOCKER_ENGINE_PATH}/bin/docker/${DOCKER_PKG}"
+  DOCKER_CLI_PATH="${DOCKER_ENGINE_PATH}/bin/docker/${DOCKER_CLI}"
+  DOCKER_CONTAINERD_PATH="${DOCKER_ENGINE_PATH}/bin/docker/${DOCKER_CONTAINERD}"
+  mkdir -p $DOCKER_ENGINE_PATH
+  mkdir -p "$DOCKER_ENGINE_PATH/bin/docker"
+
+  # STEP 1: Download files
+  $WGET $DOCKER_PKG_PATH $DOCKER_PKG_URL
+  $WGET $DOCKER_CLI_PATH $DOCKER_CLI_URL
+  $WGET $DOCKER_CONTAINERD_PATH $DOCKER_CONTAINERD_URL
+
+  # STEP 2: Install packages
+  dpkg -i $DOCKER_CONTAINERD_PATH
+  dpkg -i $DOCKER_CLI_PATH
+  dpkg -i $DOCKER_PKG_PATH
+
+  # Post installation
+  USER=$(grep 1000 "/etc/passwd" | cut -f 1 -d:)
+  [ -z "$USER" ] || usermod -aG docker "$USER"
+}
+
+function install_docker_compose() {
+  # URLs
+  DCMP_URL="https://github.com/docker/compose/releases/download/${STABLE_DOCKER_COMPOSE_VERSION}/docker-compose-$(uname -s)-$(uname -m)"
+
+  # Paths
+  DCMP_PATH="/usr/local/bin/docker-compose"
+  mkdir -p $DCMP_PATH
+
+  # STEP 1: Download file
+  $WGET $DCMP_PATH $DCMP_URL
+
+  # Post installation
+  chmod +x $DCMP_PATH
+}
+
+function post_install_check() {
+  # Docker-engine post installation check
+  if [ $INSTALL_OPTION == "engine" ]; then
+    DAPPMANAGER_CONTAINER="DAppNodeCore-dappmanager.dnp.dappnode.eth"
+
+    # If is container restarting wait 10 s
+    RESTARTING=$(docker inspect --format="{{.State.Restarting}}" $DAPPMANAGER_CONTAINER)
+    if [ $RESTARTING = true ]; then
+      sleep 10
```

Comment:
Why 10 seconds? Could it retry a few more times, like for 6 times check status and wait 10 seconds if not running

Reply:
My concern is that if the dappmanager is not running then the user will not have access to the UI, and could start doing crazy things if does not get response from the dappnode.